core/shared/src/main/scala/cats/effect/IO.scala

```diff
@@ -84,6 +84,11 @@ sealed abstract class IO[+A] private () extends IOPlatform[A] {
 
   def evalOn(ec: ExecutionContext): IO[A] = IO.EvalOn(this, ec)
 
+  def startOn(ec: ExecutionContext): IO[FiberIO[A @uncheckedVariance]] = start.evalOn(ec)
+
+  def backgroundOn(ec: ExecutionContext): ResourceIO[IO[OutcomeIO[A @uncheckedVariance]]] =
+    background.map(f => f.evalOn(ec))
```

Comment:
Ah! This is subtle but actually wrong. It will take the current `IO[A]` and run it *on the current pool* and then run the `join` on the specified pool. I believe you want the following.

```suggestion
    evalOn(ec).background
```

Reply:
A more accurate suggestion:

```suggestion
    Resource.make(start.evalOn(ec))(_.cancel).map(_.join)
```

Reply:
Okay, sounds cool. Will fix.